www/docs/server/server-side-calls.md

```diff
@@ -20,6 +20,79 @@ You may need to call your procedure(s) directly from the same server they're hos
 
 ## Create caller
 
+With the `t.createCallerFactory`-function you can create a server-side caller of any router. You first call `createCallerFactory` with an argument of the router you want to call, then this returns a function where you can pass in a `Context` for the following procedure calls.
+
+### Example
+
+We create the router with a query to list posts and a mutation to add posts, and then we a call each method.
+
+```ts twoslash
+// @target: esnext
+import { initTRPC } from '@trpc/server';
+import { z } from 'zod';
+
+type Context = {
+  foo: string;
+};
+
+const t = initTRPC.context<Context>().create();
+
+const publicProcedure = t.procedure;
+const { createCallerFactory, router } = t;
+
+interface Post {
+  id: string;
+  title: string;
+}
+const posts: Post[] = [
+  {
+    id: '1',
+    title: 'Hello world',
+  },
+];
+const appRouter = router({
+  post: router({
+    add: publicProcedure
+      .input(
+        z.object({
+          title: z.string().min(2),
+        }),
+      )
+      .mutation((opts) => {
+        const post: Post = {
+          ...opts.input,
+          id: `${Math.random()}`,
+        };
+        posts.push(post);
+        return post;
+      }),
+    list: publicProcedure.query(() => posts),
+  }),
+});
+
+// 1. create a caller-function for your router
+const createCaller = createCallerFactory(appRouter);
```

Comment:
Maybe this could have a nicer name...

```ts
const createCaller = createCallerFactory(appRouter);
const createCaller = createCallerFor(appRouter);
const createCaller = createCallerForRouter(appRouter);
```